ArduinoCode/ArduinoCode.ino

```diff
@@ -225,24 +238,33 @@ void UpdateGripper() {
 double RotateFeedback() {
   //rotates negative until it hits the limit switch, then roates to 0 and sets zero
   //had some werid behaviour trying to set zero at -180.
-  static double current;
   if (!rotateZeroSet) {
     rotateStepper.step(-1);
     rotateLimitState = digitalRead(rotateLimitPin);
     if (rotateLimitState == LOW) {
       delay(100); //delay to reducce jerk of stepper. Smooth velocity curve would be better, or lost step detection.
-      rotateStepper.step(400);
+      rotateStepper.step(stepsPerRevolution / 2);
       rotateCurrentSteps = 0;
       rotateZeroSet = 1;
     }
 
   } else {
-    stepsToRotate = RotateSetpoint * stepsPerRevolution / 360 - rotateCurrentSteps;
-    rotateCurrentSteps = rotateCurrentSteps + stepsToRotate;
+    if (RotateMode == VELOCITY) {
+      if (rotateCurrentSteps > stepsPerRevolution / 2 && rotateDir == 1) {
+        RotateSetpoint = 0;
+      }
+      if (rotateCurrentSteps < -1 * stepsPerRevolution / 2 && rotateDir == -1) {
+        RotateSetpoint = 0;
+      }
+      rotateStepper.setSpeed(RotateSetpoint);
+    } else {
+      stepsToRotate = RotateSetpoint * stepsPerRevolution / 360 - rotateCurrentSteps;
+      rotateCurrentSteps = rotateCurrentSteps + stepsToRotate;
+      RotateCurrent = rotateCurrentSteps / stepsPerRevolution * 360;
+    }
   }
 
-  current = rotateCurrentSteps / 800 * 360;
-  return current;
+  return RotateCurrent;
```

Comment:
if the limit switch is low (not actuated), and its in velocity mode, what is RotateCurrent equal to?